Tropos/Models/DailyForecast.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+@objc class DailyForecast: NSObject {
+    let date: NSDate
+    let conditionsDescription: String
+    let highTemperature: Temperature
+    let lowTemperature: Temperature
+    
+    init(JSON: Dictionary<String, AnyObject>) {
```

Comment:
Great points.

> What if something goes wrong and view DailyForecast cannot be created. Do we want to display then 2 or just 1 DailyForecast?

That's what I'm thinking. After some playing around to get one to not show up I have it looking like this which seems reasonable.

<img src="https://cloud.githubusercontent.com/assets/3799709/8501575/b69945b6-215b-11e5-8ba3-01f825992b2b.png" height="200" alt="ios simulator screen shot jul 3 2015 8 14 58 am">

> Can we assume that DailyForecast is always created and there is no case when something goes wrong? Then we can use only if-let and initializer will be always returning some objects.

I don't think we can. I haven't seen a case where it doesn't work yet but it's always a possibility since we're relying on a third party API for the data. With that in mind we should definitely have some sort of fallback. I'm thinking the easiest way to handle this would be to use the null-object pattern. When it fails we can provide a null-object and it will render empty like in the screenshot I posted earlier. 

I was originally considering making the values optional but that's not really great because there's no case where this object is valid if it's missing any of them. We can't display it. With that in mind I'm thinking there should be a `NullDailyForecast` class that returns `nil` values for everything. This way we don't have to add lots of conditional logic or mess with any RAC stuff.

After some playing around I got a diff that looks like this, still needs the failable initializer stuff:

```diff
diff --git a/Tropos/Models/DailyForecast.swift b/Tropos/Models/DailyForecast.swift
index 0075d48..99c5f25 100644
--- a/Tropos/Models/DailyForecast.swift
+++ b/Tropos/Models/DailyForecast.swift
@@ -13,3 +13,10 @@ import Foundation
         self.lowTemperature = Temperature(fahrenheitValue: JSON["temperatureMin"] as! Int)
     }
 }
+
+@objc class NullDailyForecast: NSObject {
+    let date: NSDate? = .None
+    let conditionsDescription: String? = .None
+    let highTemperature: Temperature? = .None
+    let lowTemperature: Temperature? = .None
+}
diff --git a/Tropos/Models/TRWeatherUpdate.m b/Tropos/Models/TRWeatherUpdate.m
index 56ce074..dadecd5 100644
--- a/Tropos/Models/TRWeatherUpdate.m
+++ b/Tropos/Models/TRWeatherUpdate.m
@@ -54,6 +54,10 @@
 
     for (NSUInteger index = 1; index < 4; index++) {
         DailyForecast *dailyForecast = [[DailyForecast alloc] initWithJSON:currentConditionsJSON[@"daily"][@"data"][index]];
+        if (index == 3) { dailyForecast = nil; } // ONLY FOR TESTING
+        if (dailyForecast == nil) {
+            dailyForecast = (DailyForecast *)[NullDailyForecast new];
+        }
         [dailyForecasts addObject:dailyForecast];
     }
 
diff --git a/Tropos/ViewModels/TRDailyForecastViewModel.m b/Tropos/ViewModels/TRDailyForecastViewModel.m
index bb068a7..2bbd4db 100644
--- a/Tropos/ViewModels/TRDailyForecastViewModel.m
+++ b/Tropos/ViewModels/TRDailyForecastViewModel.m
@@ -30,19 +30,31 @@
     return [dateFormatter stringFromDate:self.dailyForecast.date];
 }
 
-- (UIImage *)conditionsImage
+- (nullable UIImage *)conditionsImage
 {
-    return [UIImage imageNamed:self.dailyForecast.conditionsDescription];
+    NSString *conditionsDescription = [NSString stringWithFormat:@"%@", self.dailyForecast.conditionsDescription];
+    return [UIImage imageNamed:conditionsDescription];
 }
 
-- (NSString *)highTemperature
+- (nullable NSString *)highTemperature
 {
-    return [[TRTemperatureFormatter new] stringFromTemperature:self.dailyForecast.highTemperature];
+    return [self stringFromTemperature:self.dailyForecast.highTemperature];
 }
 
-- (NSString *)lowTemperature
+- (nullable NSString *)lowTemperature
 {
-    return [[TRTemperatureFormatter new] stringFromTemperature:self.dailyForecast.lowTemperature];
+    return [self stringFromTemperature:self.dailyForecast.lowTemperature];
+}
+
+#pragma mark - Private
+
+- (NSString *)stringFromTemperature:(Temperature *)temperature
+{
+    if (temperature == nil) {
+        return @"";
+    } else {
+        return [[TRTemperatureFormatter new] stringFromTemperature:self.dailyForecast.lowTemperature];
+    }
 }
 
 @end
```

Thoughts?